src/compiler/checker.ts

```diff
@@ -20878,18 +21080,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
 
         if (!(checkMode & SignatureCheckMode.IgnoreReturnTypes)) {
-            // If a signature resolution is already in-flight, skip issuing a circularity error

```

Comment:
Note for self: This is the change I want to back out, since it doesn't impact the speculative overload resolution fixes in this PR.